test/test_browser.py

```diff
@@ -5316,6 +5316,14 @@ def test_wasmfs_opfs(self):
     self.btest_exit(test, args=args + ['-DWASMFS_SETUP'])
     self.btest_exit(test, args=args + ['-DWASMFS_RESUME'])
 
+  @parameterized({
+    'default': (['-sCASE_INSENSITIVE_FS=1'],),
+    'wasmfs': (['-sWASMFS', '-sCASE_INSENSITIVE_FS=1', '-DWASMFS'],),
+  })
+  def test_path_icase(self, args):
```

Comment:
All the changes are in backend-independent code, so it should be fine to just test the in-memory backend. It would be good to add more testing for things like opening the same file multiple times with different capitalizations.

Reply:
I think fetch backend is already case-insensitive. Because URLs are case-insensitive.

But what about other backends, such as node backend? I don't know. It seems for Node.js, it depends on the host platform FS. And using `-sCASE_INSENSITIVE-FS` for Node.js with case sensitive FS makes little sense.
https://nodejs.org/en/docs/guides/working-with-different-filesystems/

Anyway, if there will be some problems, they can be fixed separately.

Reply:
Hmm, it seems that doing the totally correct thing on Node would involve detecting case sensitivity at runtime on a per-directory basis. Otherwise when running in case-sensitive mode on top of a case-insensitive file system, it would be possible to get multiple file handles to the same underlying file, and when running in case-insensitive mode on top of a case-sensitive file system, there could be files that couldn't be opened. A simpler fix might be to keep only mounted directories in the dcache for node directories and have the node backend handle all name resolution and manage `File` identity.

But handling node case sensitivity correctly is a little different from your use case here, right? Getting node right is about matching the underlying file system behavior, but what you're trying to do is update WasmFS to provide a case-insensitive interface so that applications above it that expect case insensitivity still work.

What do you expect should happen if there are two IDBFS files `foo.txt` and `FOO.TXT` and WasmFS is started in case-insensitive mode? Or should that never happen?

Reply:
> Hmm, it seems that doing the totally correct thing on Node would involve detecting case sensitivity at runtime on a per-directory basis.

I think adding a check at the time the backend is mounted should be enough.

> A simpler fix might be to keep only mounted directories in the dcache for node directories and have the node backend handle all name resolution and manage File identity.

Does it work differently now? Sorry, I haven't used node backend yet.

> But handling node case sensitivity correctly is a little different from your use case here, right? 
> but what you're trying to do is update WasmFS to provide a case-insensitive interface so that applications above it that expect case insensitivity still work.

That's right.

> What do you expect should happen if there are two IDBFS files foo.txt and FOO.TXT and WasmFS is started in case-insensitive mode?

If IDBFS mounts all the keys it has in the DB, then I'd expect it should mount the first file `foo.txt` and then it should ignore the second file `FOO.TXT` because such file node already exists.